src/test/java/com/commercetools/sync/cartdiscounts/utils/CartDiscountUpdateActionUtilsTest.java

```diff
@@ -0,0 +1,1473 @@
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.getName()).thenReturn(LocalizedString.of(Locale.ENGLISH, "name"));
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.getName()).thenReturn(null);
+
+        final Optional<UpdateAction<CartDiscount>> changeNameUpdateAction =
+                buildChangeNameUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeNameUpdateAction).contains(ChangeName.of(null));
+    }
+
+    @Test
+    void buildChangeNameUpdateAction_WithJustOldNameHasNullValue_ShouldBuildUpdateAction() {
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.isActive()).thenReturn(null);
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        final LocalizedString newName = LocalizedString.of(Locale.ENGLISH, "name");
+        when(newCartDiscountDraft.getName()).thenReturn(newName);
+
+        final Optional<UpdateAction<CartDiscount>> changeNameUpdateAction =
+                buildChangeNameUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeNameUpdateAction).contains(ChangeName.of(newName));
+    }
+
+    @Test
+    void buildChangeNameUpdateAction_WithBothNameHaveNullValues_ShouldNotBuildUpdateAction() {
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.getName()).thenReturn(null);
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.getName()).thenReturn(null);
+
+        final Optional<UpdateAction<CartDiscount>> changeNameUpdateAction =
+                buildChangeNameUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeNameUpdateAction).isNotPresent();
+    }
+
+    @Test
+    void buildSetDescriptionUpdateAction_WithDifferentValues_ShouldBuildUpdateAction() {
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.getDescription())
+                .thenReturn(LocalizedString.of(Locale.ENGLISH, "cart-discount-1-desc"));
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        final LocalizedString description = LocalizedString.of(Locale.ENGLISH, "new-description");
+        when(newCartDiscountDraft.getDescription()).thenReturn(description);
+
+        final Optional<UpdateAction<CartDiscount>> setDescriptionUpdateAction =
+                buildSetDescriptionUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(setDescriptionUpdateAction).contains(SetDescription.of(description));
+    }
+
+    @Test
+    void buildSetDescriptionUpdateAction_WithSameValues_ShouldNotBuildUpdateAction() {
+        final LocalizedString description = LocalizedString.of(Locale.ENGLISH, "cart-discount-1-desc");
```

Comment:
Please add a test for testing same values of different instances (which would be the more common case actually happening)